installer/tue-install-impl.bash

```diff
@@ -252,34 +267,36 @@ function tue-install-git
                 tue-install-info "URL has switched to $repo"
             fi
 
-            tue-install-debug "git -C $targetdir pull --ff-only --prune"
+            tue-install-debug "git -C $targetdir pull --ff-only --prune --recurse-submodules"
 
             local res
-            res=$(git -C "$targetdir" pull --ff-only --prune 2>&1)
+            res=$(git -C "$targetdir" pull --ff-only --prune --recurse-submodules 2>&1)
 
             tue-install-debug "$res"
 
             TUE_INSTALL_GIT_PULL_Q+=$targetdir
 
-            if [[ $res == "Already up to date"* ]]
+            if [[ $(echo "$res" | grep -v "Fetching submodule") == "Already up to date." ]]
             then
                 res=
             fi
         fi
     fi
 
     tue-install-debug "Desired version: $version"
-    if [ -n "$version" ];
+    if [ -n "$version" ]
     then
+        local _try_branch_res
         _try_branch_git "$targetdir" "$version"
-        res="$res $_try_branch_res"
+        [ -n "$_try_branch_res" ] && res="$res $_try_branch_res"
     fi
 
     tue-install-debug "Desired branch: $BRANCH"
-    if [ -n "$BRANCH" ]; #Cannot be combined with version-if because this one might not exist
+    if [ -n "$BRANCH" ] #Cannot be combined with version-if because this one might not exist
     then
+        local _try_branch_res
```

Comment:
Define this only once. Clear it in the `if` block if needed

Reply:
Done